samples/Quickstart/src/Filters/QuickstartFilter.cs

```diff
@@ -9,55 +19,89 @@
 using Microsoft.Extensions.Logging;
 using Microsoft.Extensions.Options;
 using Newtonsoft.Json.Linq;
-using System;
-
-using System.Threading.Tasks;
 
 namespace Quickstart.Filters
 {
     public class QuickstartFilter : IInputFilter
     {
-        public QuickstartFilter(IOptions<QuickstartOptions> options, IAuthenticator authenticator, TelemetryClient telemetryClient = null, ILogger<QuickstartFilter> logger = null)
+        public QuickstartFilter(IHttpCustomHeaderCollection customHeader, TelemetryClient telemetryClient = null, ILogger<QuickstartFilter> logger = null)
```

Comment:
You do not need to inject IHttpCustomHeaderCollection in the IFilter b/c you are using the RestBinding.  The RestBinding will take care of Custom Headers by itself.

Reply:
Removed as suggested. please verify.